diff --git a/node_modules/@docusaurus/theme-live-codeblock/src/theme/ReactLiveScope/index.js b/node_modules/@docusaurus/theme-live-codeblock/src/theme/ReactLiveScope/index.js
index f971d6a..631f815 100644
--- a/node_modules/@docusaurus/theme-live-codeblock/src/theme/ReactLiveScope/index.js
+++ b/node_modules/@docusaurus/theme-live-codeblock/src/theme/ReactLiveScope/index.js
@@ -6,11 +6,15 @@
  */
 
 import React from 'react';
+import { Box, NativeBaseProvider, extendTheme } from 'native-base';
 
 // Add react-live imports you need here
 const ReactLiveScope = {
   React,
   ...React,
+  Box,
+  NativeBaseProvider,
+  extendTheme,
 };
 
 export default ReactLiveScope;
