diff --git a/node_modules/@docusaurus/core/lib/webpack/base.js b/node_modules/@docusaurus/core/lib/webpack/base.js
index 567ed59..586d7f1 100644
--- a/node_modules/@docusaurus/core/lib/webpack/base.js
+++ b/node_modules/@docusaurus/core/lib/webpack/base.js
@@ -66,9 +66,10 @@ function createBaseConfig(props, isServer, minify = true) {
         },
         devtool: isProd ? false : 'cheap-module-eval-source-map',
         resolve: {
-            extensions: ['.wasm', '.mjs', '.js', '.jsx', '.ts', '.tsx', '.json'],
+            extensions: ['.wasm', '.mjs', '.web.js', '.web.jsx', '.web.ts', '.web.tsx' ,'.js', '.jsx', '.ts', '.tsx', '.json'],
             symlinks: true,
-            alias: Object.assign({ '@site': siteDir, '@generated': generatedFilesDir }, getDocusaurusAliases()),
+            alias: Object.assign({ '@site': siteDir, '@generated': generatedFilesDir, 'react-native$': path_1.default.resolve(siteDir,'node_modules','react-native-web'),
+          }, getDocusaurusAliases()),
             // This allows you to set a fallback for where Webpack should look for modules.
             // We want `@docusaurus/core` own dependencies/`node_modules` to "win" if there is conflict
             // Example: if there is core-js@3 in user's own node_modules, but core depends on
