diff --git a/node_modules/@docusaurus/mdx-loader/src/index.js b/node_modules/@docusaurus/mdx-loader/src/index.js
index 5023a09..5ad2a76 100644
--- a/node_modules/@docusaurus/mdx-loader/src/index.js
+++ b/node_modules/@docusaurus/mdx-loader/src/index.js
@@ -5,8 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-const {getOptions} = require('loader-utils');
-const {readFile} = require('fs-extra');
+const { getOptions } = require('loader-utils');
+const { readFile } = require('fs-extra');
 const mdx = require('@mdx-js/mdx');
 const emoji = require('remark-emoji');
 const matter = require('gray-matter');
@@ -24,7 +24,7 @@ const DEFAULT_OPTIONS = {
 module.exports = async function docusaurusMdxLoader(fileString) {
   const callback = this.async();
 
-  const {data, content} = matter(fileString);
+  const { data, content } = matter(fileString);
   const reqOptions = getOptions(this) || {};
   const options = {
     ...reqOptions,
@@ -33,13 +33,20 @@ module.exports = async function docusaurusMdxLoader(fileString) {
       ...DEFAULT_OPTIONS.remarkPlugins,
       [
         transformImage,
-        {staticDir: reqOptions.staticDir, filePath: this.resourcePath},
+        { staticDir: reqOptions.staticDir, filePath: this.resourcePath },
       ],
       [
         transformLinks,
-        {staticDir: reqOptions.staticDir, filePath: this.resourcePath},
+        { staticDir: reqOptions.staticDir, filePath: this.resourcePath },
       ],
-      ...(reqOptions.remarkPlugins || []),
+      ...(reqOptions.remarkPlugins.map((r) => [
+        r,
+        {
+          staticDir: reqOptions.staticDir,
+          filePath: this.resourcePath,
+          directory: this.context,
+        },
+      ]) || []),
     ],
     rehypePlugins: [
       ...(reqOptions.beforeDefaultRehypePlugins || []),
